addons/mail/static/src/models/user_setting/user_setting.js

```diff
@@ -0,0 +1,321 @@
+/** @odoo-module **/
+
+import { browser } from "@web/core/browser/browser";
+
+import { registerNewModel } from '@mail/model/model_core';
+import { attr, one2one, one2many } from '@mail/model/model_field';
+import { create } from '@mail/model/model_field_command';
+
+function factory(dependencies) {
+
+    class UserSetting extends dependencies['mail.model'] {
+
+        /**
+         * @override
+         */
+        _created() {
+            const res = super._created(...arguments);
+            this._timeoutIds = {};
+            this._loadLocalSettings();
+            return res;
+        }
+
+        /**
+         * @override
+         */
+        _willDelete() {
+            for (const timeoutId of Object.values(this._timeoutIds)) {
+                browser.clearTimeout(timeoutId);
+            }
+            return super._willDelete(...arguments);
+        }
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @static
+         * @param {Object} data
+         * @returns {Object}
+         */
+        static convertData(data) {
+            const data2 = {};
+            if ('use_push_to_talk' in data) {
+                data2.usePushToTalk = data.use_push_to_talk;
+            }
+            if ('push_to_talk_key' in data) {
+                data2.pushToTalkKey = data.push_to_talk_key || '';
+            }
+            if ('voice_active_duration' in data) {
+                data2.voiceActiveDuration = data.voice_active_duration;
+            }
+            if ('id' in data) {
+                data2.id = data.id;
+            }
+            return data2;
+        }
+
+        /**
+         * @returns {Object} MediaTrackConstraints
+         */
+        getAudioConstraints() {
+            const constraints = {
+                echoCancellation: true,
+                noiseSuppression: true,
+            };
+            if (this.audioInputDeviceId) {
+                constraints.deviceId = this.audioInputDeviceId;
+            }
+            return constraints;
+        }
+
+        /**
+         * @param {event} ev
+         * @param {Object} param1
+         * @param {boolean} param1.ignoreModifiers
+         */
+        isPushToTalkKey(ev, { ignoreModifiers = false } = {}) {
```

Comment:
The method name misses the aspect "pressed".

I suppose you feel it becomes hard to read. The method name is not specially too long according to amount of characters, but there are many words so it requires some effort to read and get it.

I feel it would be appropriate to use abbreviation form for "Push-To-Talk" to "PTT", and use this abbreviation only in part of code with the context/domain that is clearly webRTC related.

Unfortunately current code split is not yet by domain/context/feature but instead split by module/structure. Here `user_setting` is too broad, so abbreviations like PTT should be avoided.

Reply:
I viewed it more like a method to check if the event contained the right key, as it is called both when pressed and released, a more complex version of `ev.key === pushToTalkKey`, maybe it shouldn't take an event as argument.

Reply:
Ok I get it, let's keep the name as is then

It should be more a method of `Event` model if we had it and we follow the "by module/structure" split of code.
But let's not change method name and location in this PR.